atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/service/SplitKeyDelegatingTransactionServiceTest.java

```diff
@@ -166,6 +167,52 @@ public void ignoreUnknownFailsIfSeeingATimestampServiceItDoesNotRecognizeForMult
         Mockito.verifyNoMoreInteractions(delegate1);
     }
 
+    @Test
+    public void canMarkInProgress() {
+        assertThatCode(() -> lastDigitFiveImpliesUnknownTransactionService.markInProgress(1L))
+                .doesNotThrowAnyException();
+
+        ArgumentCaptor<Long> captor = ArgumentCaptor.forClass(Long.class);
+        verify(delegate1).markInProgress(captor.capture());
+        assertThat(captor.getValue()).isEqualTo(1L);
+    }
+
+    @Test
+    public void canMarkInProgressMultiple() {
+        assertThatCode(() -> lastDigitFiveImpliesUnknownTransactionService.markInProgress(ImmutableList.of(1L, 2L)))
+                .doesNotThrowAnyException();
+
+        ArgumentCaptor<Long> captor = ArgumentCaptor.forClass(Long.class);
+
+        verify(delegate1).markInProgress(captor.capture());
+        assertThat(captor.getValue()).isEqualTo(1L);
+
+        verify(delegate2).markInProgress(captor.capture());
+        assertThat(captor.getValue()).isEqualTo(2L);
+    }
+
+    @Test
+    public void failsIfSeeingATimestampServiceItDoesNotRecognizeForMarkInProgress() {
+        assertThatLoggableExceptionThrownBy(() -> lastDigitFiveImpliesUnknownTransactionService.markInProgress(7L))
+                .isInstanceOf(SafeIllegalStateException.class)
+                .hasMessageContaining("Could not find a transaction service for the given timestamp");
```

Comment:
Didn't look at the impl, but it seems a bit strange that `lastDigitFiveImpliesUnknownTransactionService` throws on a _seven_. I know that the inverse is not logically equivalent, but this still reads confusingly.